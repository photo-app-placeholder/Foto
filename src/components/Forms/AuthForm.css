.auth {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: auto;
  width: fit-content;
}

.auth input {
  width: 210px;
  height: 37px;
  margin: 10px;
  font-size: 12pt;
}

.auth button {
  width: fit-content;
  justify-content: center;
  width: 218px;
  height: 43px;
  margin: 8px 12px 10px 12px;
  font-size: 12pt;
}

.terms {
  width: 230px;
  font-size: 9pt;
  color: rgb(68, 67, 67);
}
