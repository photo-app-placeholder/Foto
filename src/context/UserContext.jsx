import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

export const UserContext = createContext();

export const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  console.log(user);

  const contextValue = { user, setUser };

  return (
    <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>
  );
};

export const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('Error: useUser needs to be used within UserContext');
  }
  return context;
};
