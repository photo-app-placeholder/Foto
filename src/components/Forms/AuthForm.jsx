import React, { useState } from 'react';
import { createProfile } from '../../services/profiles';
import { signInUser, signUpUser } from '../../services/users';
import styles from './AuthForm.css';
import { useUser } from '../../context/UserContext';
import { useHistory } from 'react-router-dom';
import { Link } from 'react-router-dom/cjs/react-router-dom.min';

const { auth, terms } = styles;
export default function AuthForm({ isRegistering }) {
  const [formState, setFormState] = useState({
    email: '',
    username: '',
    password: '',
  });
  const { setUser } = useUser();
  const history = useHistory();

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      if (isRegistering) {
        let resp = await signUpUser(formState.email, formState.password);
        console.log(resp);
        await createProfile({
          username: formState.username,
          bio: '',
          user_id: resp.id,
        });
        alert('Email confirmation sent, confirm email address then log in.');
        history.push('/login');
      } else {
        let resp = await signInUser(formState.email, formState.password);
        setUser(resp);
        window.location.replace(`/${formState.username}`);
      }
    } catch (error) {
      alert(`Login failed, please try again. error: ${error}`);
    }
  };

  return (
    <form className={auth} onSubmit={handleSubmit}>
      <input
        type="email"
        placeholder="Email"
        value={formState.email}
        onChange={(e) => setFormState({ ...formState, email: e.target.value })}
      />
      {isRegistering && (
        <>
          <input
            type="text"
            placeholder="Username"
            value={formState.username}
            onChange={(e) =>
              setFormState({ ...formState, username: e.target.value })
            }
          />
        </>
      )}

      <input
        type="password"
        placeholder="Password"
        value={formState.password}
        onChange={(e) =>
          setFormState({ ...formState, password: e.target.value })
        }
      />
      {isRegistering && (
        <p className={terms}>
          By signing up, you agree not to upload photos that contain graphic or
          disturbing imagery that may be against the law.
        </p>
      )}
      <button type="submit">{isRegistering ? 'Sign Up' : 'Login'}</button>
      {isRegistering ? (
        <p>
          Already have an account? <Link to="/login">Login</Link>
        </p>
      ) : (
        <p>
          New? <Link to="/register">Register</Link>
        </p>
      )}
    </form>
  );
}
