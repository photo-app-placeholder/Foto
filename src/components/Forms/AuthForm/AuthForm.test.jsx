import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../../context/UserContext';
import AuthForm from './AuthForm';

test('should display a form to add images', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <AuthForm />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
