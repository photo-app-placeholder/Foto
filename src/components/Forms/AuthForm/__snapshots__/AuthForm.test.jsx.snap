// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a form to add images 1`] = `
<div>
  <form
    class="auth"
  >
    <input
      placeholder="Email"
      type="email"
      value=""
    />
    <input
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      Login
    </button>
    <p>
      New? 
      <a
        href="/register"
      >
        Register
      </a>
    </p>
  </form>
</div>
`;
